### PR TITLE
fix(@schematics/update): allow --from and --to to have partial versions

### LINKED ISSUE #660: Allow any shorthand version for the `--from` argument on @schematics/update:migrate
* `1` should be equivalent to `1.0.0`
* `1.1` should be equivalent to `1.1.0`

### PR DESCRIPTION
**Pull request by hansl**
Basically, --from=1 is the same as --from=1.0.0.

Fix #660

### CODE REVIEW

File: packages/schematics/update/update/index_spec.ts [hunk 1]
```diff
@@ -153,4 +153,37 @@ describe('@schematics/update', () => {
...
+
+    schematicRunner.runSchematicAsync('update', {
+      packages: ['@angular-devkit-tests/update-migrations'],
+      migrateOnly: true,
+      from: '0',
+    }, appTree).pipe(
+      map(tree => {
+        const packageJson = JSON.parse(tree.readContent('/package.json'));
+        expect(packageJson['dependencies']['@angular-devkit-tests/update-migrations'])
+          .toBe('1.6.0');
```
Comment (filipesilva) [inline packages/schematics/update/update/index_spec.ts]: I don't follow the test. Before running the schematic you set the version, then you run the schematic and check the version hasn't changed.

File: packages/schematics/update/update/index_spec.ts [hunk 2]
```diff
@@ -153,4 +153,37 @@ describe('@schematics/update', () => {
...
+
+        // Check install task.
+        expect(schematicRunner.tasks).toEqual([
+          {
+            name: 'run-schematic',
+            options: jasmine.objectContaining({
+              name: 'migrate',
```
Comment (filipesilva) [inline packages/schematics/update/update/index_spec.ts]: I think this should be checking the install task, like some of the other tests in this file.

File: packages/schematics/update/update/index_spec.ts (hunk 1 above)
Comment (hansl) [reply packages/schematics/update/update/index_spec.ts]: Yes, it's a migration only update.

File: packages/schematics/update/update/index_spec.ts (hunk 2 above)
Comment (hansl) [reply packages/schematics/update/update/index_spec.ts]: There's no installation in a migration only.

**Pull request merged**